lib/extract-templates.js

```diff
@@ -0,0 +1,94 @@
+// import { getTemplateLocals } from '@glimmer/syntax';
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+// import { preprocessEmbeddedTemplates } from 'ember-template-imports/lib/preprocess-embedded-templates.js';
+// import * as util from 'ember-template-imports/src/util.js';
+
+const extensions = ['js', 'ts', 'gjs', 'gts'];
+/**
+ * @typedef {object} TemplateInfo
+ * @property {number} line
+ * @property {number} column
+ * @property {string} template
+ *
+ *
+ * @param {string} moduleSource
+ * @param {string} relativePath
+ *
+ * @returns {TemplateInfo[]}
+ */
+export function extractTemplates(moduleSource, relativePath) {
+  let mightHaveTemplates = extensions.some((ext) => relativePath.endsWith(ext));
+
+  if (!mightHaveTemplates) {
+    return [{ line: 0, column: 0, template: moduleSource }];
+  }
+
+  let result = [];
+
+  // const preprocessed = preprocessEmbeddedTemplates(moduleSource, {
```

Comment:
all this does is swap out `<template>` for the vanilla JS variant (which is more appropriate for eslint-plugin-ember in https://github.com/ember-cli/eslint-plugin-ember/pull/1395)